python/dgl/graphbolt/graph_storage/csc_sampling_graph.py

```diff
@@ -173,6 +173,26 @@ def metadata(self) -> Optional[GraphMetadata]:
         """
         return self._metadata
 
+    def in_subgraph(self, nodes: torch.Tensor) -> torch.ScriptObject:
+        """Return the subgraph induced on the inbound edges of the given nodes.
+
+        An in subgraph is equivalent to creating a new graph using the incoming
+        edges of the given nodes.
+
+        Parameters
+        ----------
+        nodes : torch.Tensor
+            The nodes to form the subgraph which are type agnostic.
+
+        Returns
+        -------
+        SampledSubgraph
+            The in subgraph.
+        """
+        # Remove duplicate nodes.
+        nodes = torch.unique(nodes)
```

Comment:
In which case user will call in_subgraph with duplicated nodes? Is it a valid query?

Reply:
just in case of user pass in duplicate node in mistake.

Reply:
If it is a user mistake, let's catch it and throw an error rather than fix it silently.

Reply:
updated.